### DISCUSSION TITLE
How to skip ignore certain files for for routes generation, but still have file watcher on them?

### DISCUSSION THREAD

**Discussion by tribe-kp** [nuxt/nuxt#17659] [Questions]
I want to have page specific helper components near the route file, but then they are generated as the separate routes, example:
![image](https://user-images.githubusercontent.com/100837313/156537154-f6ec0497-207a-41e3-984e-f0657740f46b.png)

If I add `ignore: ['**/_components']` to `.nuxt.config.ts`, these helper components are skipped for routes generation, but they are also skipped form file watcher. So any file change is not catched via dev server. Is it possible to solve this?

**Comment by Megarsh** [comment]
You should really use the components directory for that https://v3.nuxtjs.org/docs/directory-structure/components.

For your convenience you can create folders inside of it that match your pages so for example if you create a  ```components/index/Documentations.vue``` component you can use it inside of index.vue as ```<IndexDocumentations />```

**Reply by tribe-kp** [reply]
But it creates duplication of the routes structure and separates related components

**Reply by zhoutengshen** [reply]
> You should really use the components directory for that https://v3.nuxtjs.org/docs/directory-structure/components.
> 
> For your convenience you can create folders inside of it that match your pages so for example if you create a `components/index/Documentations.vue` component you can use it inside of index.vue as `<IndexDocumentations />`

For some business logic, I hope to abide by the proximity principle, because there are many routes in the project, and the proximity principle can facilitate management and search for files

**Reply by VictorRibeiroKCX** [reply]
@zhoutengshen yes, in some cases we just want a local component that will only be used in that specific view context. Being as near as possible in a deeply nested page folder just seems better.

**Comment by lucassimines** [comment]
I need that too

**Comment by florian-lefebvre** [comment]
I think you should look at the following implementation: https://github.com/areindl/nuxt-structure-with-modules/tree/nuxt3

**Reply by VictorRibeiroKCX** [reply]
Broken link 😭

**Comment by hay-kot** [comment]
I've been trying to figure this out for some time, the only solution I was able to come up with was a applying the ignore rule only in production, which is less than idea. 

```js
// https://v3.nuxtjs.org/api/configuration/nuxt.config
const DEVELOPMENT = process.env.NODE_ENV === 'development';

export default defineNuxtConfig({
  ignore: [
    DEVELOPMENT ? '' : 'pages/**/[A-Z]*.vue',
    DEVELOPMENT ? '' : 'pages/**/*.js',
  ],
  ignoreOptions: {
    ignorecase: false,
  },
});
```

It would be nice to have a watchIgnore list (or something), where it would ignore the implications of the directory, but still be watched by Vite _or_ some other form of first class support for this type of structure.
